Tools/CMake/Patch.cmake

```diff
@@ -0,0 +1,362 @@
+# This script patches and signs all the libraries found in the PATH variable. In addition,
+# for every library that it patches, it creates an empty file, logging its action. So, for
+# `R_HOME/lib/libR.dylib`, we will have `R_HOME/lib/libR.dylib.patched.log`. I'm basically
+# implementing a simple cache to not patch things several times. This is especially useful
+# when we are installing the modules
+#
+# Notes:
+#
+#   - While this can easily be a CMake Function, or a Macro, it is not,
```

Comment:
They really don't have to be, I don't want them to be linked, because I want the  `otoolstuff.h/cpp` to replace the most immediate path to the `R_HOME/lib` and that's good enough while building, and installing packages, and then the `Patch.R` does the rest.

At the moment, what they share is those dangling libraries, e.g., `libtbbmalloc.dylib`, which I'm going to put them all into a file, and make CMake and `otoolstuff` read them from there that we don't have to sync them. I have it in my todo list already...